### PR TITLE
Add support for local product search

### PR DESCRIPTION
**Pull request by malinajirka**
This PR adds support for quering products by a searchQuery from the local database. It is used in [this PR in the woocommerce repo](https://github.com/woocommerce/woocommerce-android/pull/8391).

To Test:
[More on WCAndroid repo PR.](https://github.com/woocommerce/woocommerce-android/pull/8391)

### CODE REVIEW

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/ProductSqlUtils.kt [hunk 1]
```diff
@@ -199,7 +199,14 @@ object ProductSqlUtils {
             queryBuilder.contains(WCProductModelTable.CATEGORIES, categoryFilter)
         }
         if (searchQuery?.isNotEmpty() == true) {
-            queryBuilder.contains(WCProductModelTable.NAME, searchQuery)
+            queryBuilder
```
Comment (kidinov) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/ProductSqlUtils.kt:202]: I see there is a test of this class. Maybe to validate this code there as well?

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/ProductSqlUtils.kt [hunk 2]
```diff
@@ -176,7 +176,8 @@ object ProductSqlUtils {
         site: SiteModel,
```
Comment (kidinov) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/ProductSqlUtils.kt:176]: np: maybe to put in the name that search query is also accepted by the method?

**Review by kidinov — APPROVED** [review]
LGTM! I left a couple of not blocking comments

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/ProductSqlUtils.kt (hunk 1 above)
Comment (malinajirka) [reply plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/ProductSqlUtils.kt:202]: I missed that, thanks! Added in 89f70ebbe923fe64fc50d28081b9f12d24372882

**Comment by malinajirka** [comment]
Thanks so much for the review @kidinov ! I've implemented both suggestions and requested a re-review in case you'd have some feedback on the method name or the test method. Thanks 🙇

**Pull request merged**